### PR TITLE
Resolve JSON files from extensionless parameters

### LINKED ISSUE #19: opts.extensions should include '.json'
Node's `require()` can resolve JSON files without passing the extension.

### PR DESCRIPTION
**Pull request by SLaks**
Fixes #19

### CODE REVIEW

**Comment by alexisvincent** [comment]
+1

**Comment by virtulis** [comment]
wonder if there's a reason this wasn't merged :(

File: lib/sync.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function (x, opts) {
     };
     var readFileSync = opts.readFileSync || fs.readFileSync;
     
-    var extensions = opts.extensions || [ '.js' ];
+    var extensions = opts.extensions || [ '.js', '.json' ];
```
Comment (ljharb) [inline lib/sync.js]: This is missing `.node` - instead, could we use the keys in `require.extensions`, which should ensure that the array always matches the require algorithm?

(if there's a concern that browserify won't be able to handle `require.extensions`, then let's add `.node` to the list also)

File: test/resolver_sync.js [hunk 2]
```diff
@@ -21,6 +21,26 @@ test('foo', function (t) {
     t.end();
 });
 
+test('JSON', function (t) {
+    var dir = __dirname + '/resolver';
```
Comment (ljharb) [inline test/resolver_sync.js]: this should use `path.join(__dirname, 'resolver')` instead

**Comment by ljharb** [comment]
Closing this in favor of either #145 or #138 (my preference of the two).

**Pull request closed without merging**